app/server/lib/DocApi.ts

```diff
@@ -871,6 +872,26 @@ export class DocWorkerApi {
       res.json(result);
     }));
 
+    this._app.get('/api/docs/:docId/download/table-schema', canView, withDoc(async (activeDoc, req, res) => {
+      const doc = await this._dbManager.getDoc(req);
+      const options = this._getDownloadOptions(req, doc.name);
+      const tableSchema = await collectTableSchemaInFrictionlessFormat(activeDoc, req, options);
+      const apiPath = await this._grist.getResourceUrl(doc, 'api');
+      const query = req.originalUrl.split('?', 2)[1];
```

Comment:
Hmm technically there could be literal `?` characters in query parameters, and the split limit doesn't quite do the right thing in this case. How about something more like the second example in https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams work?